Makefile-ponyc

```diff
@@ -232,6 +232,14 @@ define USE_CHECK
     else
       $$(warning LLVM_LINK_STATIC already set to '$(LLVM_LINK_STATIC)'; using pre-existing value)
     endif
+  else ifeq ($1,memtrack)
+    ALL_CFLAGS += -DUSE_MEMTRACK
+    ALL_CXXFLAGS += -DUSE_MEMTRACK
+    PONY_BUILD_DIR := $(PONY_BUILD_DIR)-memtrack
+  else ifeq ($1,memtrack_messages)
```

Comment:
this feels a little odd that memtrack_messages turns on both. given that it's two different defines.

Reply:
understood. i can completely separate them if you'd like.

Reply:
Would there be value to memtrack_messages without memtrack?

Reply:
my thinking was that `memtrack_messages` built upon `memtrack` but it is possible others may find value in `memtrack_messages` without the base tracking provided by `memtrack`. To clarify, my thought process was that someone might want the memory tracking provided by `memtrack` only due to performance concerns with having the full tracking provided by `memtrack_messages` due to the per message send/receive bookkeeping given that there's usually lots of messages being sent/received.

Reply:
This makes sense so leaving as is seems reasonable.